server/src/main/java/org/elasticsearch/script/Metadata.java

```diff
@@ -172,7 +119,7 @@ public void setVersion(long version) {
     }
 
     public ZonedDateTime getTimestamp() {
```

Comment:
It is available in 2 of 4 write contexts, ingest and update.  It is missing in reindex and update by query

It's here because that allows scripts to do `Metadata m = metadata(); m.timestamp`.  Otherwise, the script would have to name the subtype.

It is possible to add timestamp to reindex and update by query, both have access to the thread pool and pull a long supplier from it.

Reply:
We could duck-type the Metadata subclasses just for timestamp.

Reply:
If it’s relevant to all the other contexts, then it makes sense to stay completely in this class. But I don’t think we should have it defined here but implemented in subclasses.

Reply:
I'll change the implementation here in the Update PR.